exasol/analytics/udf/communication/peer_communicator/background_listener_interface.py

```diff
@@ -56,79 +76,75 @@ def __init__(
+            socket_factory, f"in_control_socket{id(self)}"
+        )
         self._is_ready_to_stop = False
         self._background_listener_run = BackgroundListenerThread(
             name=self._name,
             number_of_peers=number_of_peers,
             socket_factory=socket_factory,
             listen_ip=listen_ip,
             group_identifier=group_identifier,
-            out_control_socket_address=out_control_socket_address,
-            in_control_socket_address=in_control_socket_address,
+            out_control_socket_address=self._out_control.address,
+            in_control_socket_address=self._in_control.address,
             clock=clock,
             config=config,
             trace_logging=trace_logging,
         )
         self._thread = threading.Thread(target=self._background_listener_run.run)
         self._thread.daemon = True
         self._thread.start()
-        self._set_my_connection_info()
-
-    def _create_in_control_socket(self, socket_factory: SocketFactory) -> str:
-        self._in_control_socket: Socket = socket_factory.create_socket(SocketType.PAIR)
-        in_control_socket_address = (
-            f"inproc://BackgroundListener_in_control_socket{id(self)}"
-        )
-        self._in_control_socket.bind(in_control_socket_address)
-        return in_control_socket_address
-
-    def _create_out_control_socket(self, socket_factory: SocketFactory) -> str:
-        self._out_control_socket: Socket = socket_factory.create_socket(SocketType.PAIR)
-        out_control_socket_address = (
-            f"inproc://BackgroundListener_out_control_socket{id(self)}"
-        )
-        self._out_control_socket.bind(out_control_socket_address)
-        return out_control_socket_address
+        self._my_connection_info = self._get_my_connection_info()
 
-    def _set_my_connection_info(self):
-        message = None
+    def _get_my_connection_info(self) -> ConnectionInfo:
+        received = None
         try:
-            message = self._out_control_socket.receive()
-            message_obj: messages.Message = deserialize_message(
-                message, messages.Message
-            )
-            specific_message_obj = message_obj.__root__
-            assert isinstance(specific_message_obj, messages.MyConnectionInfo)
-            self._my_connection_info = specific_message_obj.my_connection_info
+            received = self._out_control.socket.receive()
+            generic = deserialize_message(received, messages.Message)
+            message = generic.__root__
+            if not isinstance(message, messages.MyConnectionInfo):
+                raise UnexpectedMessageError(
+                    f"Unexpected message of type {type(message)}."
+                )
+            return message.my_connection_info
         except Exception as e:
-            self._logger.exception("Exception", raw_message=message)
+            self._logger.exception("Exception", raw_message=received)
```

Comment:
What if the exception happened in self._out_control.socket.receive()?
Then variable `received` is uninitialized, and thus the log would be useless.

Reply:
The original implementation used variable `message` which is 2 lines further down and hence with even more risk to be uninitialized. 
The logger is a structured logger (I can give additional explanation). 
Hence, a potentially uninitialized variable is acceptable from my perspective. 
We also added a `raise` statement after logging, which was missing before. 

All in all, I think the implementation is significantly improved and hopefully fine. 
But please only agree, if I could convince you.